### PR TITLE
FLW-713 Add "Fetch nominations history" test

### PR DESCRIPTION
**Pull request by ef1rspb**

### CODE REVIEW

File: fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBatchCall.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import FearlessUtils
+import Foundation
+import IrohaCrypto
+
+struct SubscanBatchCall {
+    let controllers: [String]
```
Comment (ef1rspb) [inline fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBatchCall.swift:6]: naming?

File: fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBondCall.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import FearlessUtils
+import Foundation
+import IrohaCrypto
+
+struct SubscanBondCall {
+    let controller: String
```
Comment (ef1rspb) [inline fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBondCall.swift:6]: naming?

File: fearless/Common/Network/Subscan/Data/Extrinsics/SubscanNominateCall.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import FearlessUtils
+import Foundation
+import IrohaCrypto
+
+struct SubscanNominateCall {
+    let controllers: [String]
```
Comment (ef1rspb) [inline fearless/Common/Network/Subscan/Data/Extrinsics/SubscanNominateCall.swift:6]: naming?

File: fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBatchCall.swift [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        guard let data = callArgs.stringValue?.data(using: .utf8) else { return nil }
+        guard let array = try? JSONDecoder().decode([InnerRepresentation]?.self, from: data) else { return nil }
+        let controllers = array
+            .map(\.value)
+            .map { wrappers -> [String] in
+                wrappers
+                    .filter { $0.call_function == .bond }
+                    .map { wraper -> [String] in
+                        wraper.call_args
+                            .filter { $0.name == "controller" && $0.type == "Address" }
```
Comment (ERussel) [inline fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBatchCall.swift:18]: Should we use the approach with SubscanExtrinsicsAccountId

File: fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBatchCall.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    let controllers: [String]
+
+    init?(callArgs: JSON, chain: Chain) {
+        guard let data = callArgs.stringValue?.data(using: .utf8) else { return nil }
+        guard let array = try? JSONDecoder().decode([InnerRepresentation]?.self, from: data) else { return nil }
+        let controllers = array
+            .map(\.value)
+            .map { wrappers -> [String] in
+                wrappers
+                    .filter { $0.call_function == .bond }
```
Comment (ERussel) [inline fearless/Common/Network/Subscan/Data/Extrinsics/SubscanBatchCall.swift:15]: I think we are missing set_controller here

File: fearlessIntegrationTests/RewardPayoutsServiceTests.swift [hunk 6]
```diff
@@ -212,6 +213,172 @@ class RewardPayoutsServiceTests: XCTestCase {
         }
     }
 
+    func testFetchNominationHistory() {
+        let subscanOperationFactory = SubscanOperationFactory()
+        let queue = OperationQueue()
+        let nominatorStachAccount = "5DEwU2U97RnBHCpfwHMDfJC7pqAdfWaPFib9wiZcr2ephSfT"
```
Comment (ERussel) [inline fearlessIntegrationTests/RewardPayoutsServiceTests.swift:219]: stash

File: fearless/Common/Network/Subscan/Data/Extrinsics/SubscanNominateCall.swift (hunk 3 above)
Comment (ERussel) [inline fearless/Common/Network/Subscan/Data/Extrinsics/SubscanNominateCall.swift:6]: validatorAddresses?

File: fearlessIntegrationTests/RewardPayoutsServiceTests.swift [hunk 7]
```diff
@@ -212,6 +213,194 @@ class RewardPayoutsServiceTests: XCTestCase {
...
+            queue: queue)
+
+        let controllers = (bondControllers + setControllers)
+            .compactMap { SubscanBondCall(callArgs: $0, chain: chain) }
+            .map { $0.controller }
+        return Set(controllers)
+    }
+
+    private func fetchControllersByUtilityModule(
+        nominatorStachAccount: String,
```
Comment (ERussel) [inline fearlessIntegrationTests/RewardPayoutsServiceTests.swift:297]: Stash

File: fearlessIntegrationTests/RewardPayoutsServiceTests.swift [hunk 8]
```diff
@@ -212,6 +213,194 @@ class RewardPayoutsServiceTests: XCTestCase {
...
+                "5C8Pev9UHEtBgd1XKhg38U4PC49Azx8v5uphtxjWiXwmpsc7"
+            ])
+            XCTAssertEqual(validatorIdsSet, expectedValidatorIdsSet)
+        } catch {
+            XCTFail("Unexpected error: \(error)")
+        }
+    }
+
+    private func fetchControllersByStakingModule(
+        nominatorStachAccount: String,
```
Comment (ERussel) [inline fearlessIntegrationTests/RewardPayoutsServiceTests.swift:271]: Stash

**Review by ERussel — APPROVED** [review]

**Pull request merged**